File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/models/SOP/SOPHistoryData.java
package egovframework.base.history.ibll.models.sop;

import java.time.LocalDateTime;
import java.util.List;

public class SOPHistoryData {

    private int rowNo = -1;
    private Integer sensorZoneHistoryNo = null;
    private int actionStepHistoryNo = -1;
    private Integer lastAccessedUserNo = null;
    private String disasterCategoryName = "";
    private String sopName = "";
    private String actionStepName = "";
    private String sensorName = null;
    private String position = "";
    private LocalDateTime beginTime = LocalDateTime.now();
    private LocalDateTime endTime = null;
    private String userName = null;
    private List<Integer> allSensorZoneNos = null;

    public int getRowNo() {
        return rowNo;
    }

    public void setRowNo(int rowNo) {
        this.rowNo = rowNo;
    }

    public Integer getSensorZoneHistoryNo() {
        return sensorZoneHistoryNo;
    }

    public void setSensorZoneHistoryNo(Integer sensorZoneHistoryNo) {
        this.sensorZoneHistoryNo = sensorZoneHistoryNo;
    }

    public int getActionStepHistoryNo() {
        return actionStepHistoryNo;
    }

    public void setActionStepHistoryNo(int actionStepHistoryNo) {
        this.actionStepHistoryNo = actionStepHistoryNo;
    }

    public Integer getLastAccessedUserNo() {
        return lastAccessedUserNo;
    }

    public void setLastAccessedUserNo(Integer lastAccessedUserNo) {
        this.lastAccessedUserNo = lastAccessedUserNo;
    }

    public String getDisasterCategoryName() {
        return disasterCategoryName;
    }

    public void setDisasterCategoryName(String disasterCategoryName) {
        this.disasterCategoryName = disasterCategoryName;
    }

    public String getSopName() {
        return sopName;
    }

    public void setSopName(String sopName) {
        this.sopName = sopName;
    }

    public String getActionStepName() {
        return actionStepName;
    }

    public void setActionStepName(String actionStepName) {
        this.actionStepName = actionStepName;
    }

    public String getSensorName() {
        return sensorName;
    }

    public void setSensorName(String sensorName) {
        this.sensorName = sensorName;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public LocalDateTime getBeginTime() {
        return beginTime;
    }

    public void setBeginTime(LocalDateTime beginTime) {
        this.beginTime = beginTime;
    }

    public LocalDateTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalDateTime endTime) {
        this.endTime = endTime;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public List<Integer> getAllSensorZoneNos() {
        return allSensorZoneNos;
    }

    public void setAllSensorZoneNos(List<Integer> allSensorZoneNos) {
        this.allSensorZoneNos = allSensorZoneNos;
    }
}

File: src/SOP8/Base/History/Base.History.IBLL/Models/SOP/SOPHistoryData.cs
using System;
using System.Collections.Generic;

namespace Base.History.IBLL.Models.SOP
{
    public class SOPHistoryData
    {
        private int m_nRowNo = -1;
        private int? m_nSensorZoneHistoryNo = null;
        private int m_nActionStepHistoryNo = -1;
        private int? m_nLastAccessedUserNo = null;
        private string m_strDisasterCategoryName = "";
        private string m_strSopName = "";
        private string m_strActionStepName = "";
        private string m_strSensorName = null;
        private string m_strPosition = "";
        private DateTime m_beginTime = new DateTime();
        private DateTime? m_endTime = null;
        private string m_strUserName = null;
        private List<int> m_allSensorZoneNos = null;

        public int RowNo
        {
            get { return m_nRowNo; }
            set { m_nRowNo = value; }
        }

        public int? SensorZoneHistoryNo
        {
            get { return m_nSensorZoneHistoryNo; }
            set { m_nSensorZoneHistoryNo = value; }
        }

        public int ActionStepHistoryNo
        {
            get { return m_nActionStepHistoryNo; }
            set { m_nActionStepHistoryNo = value; }
        }

        public int? LastAccessedUserNo
        {
            get { return m_nLastAccessedUserNo; }
            set { m_nLastAccessedUserNo = value; }
        }

        public string DisasterCategoryName
        {
            get { return m_strDisasterCategoryName; }
            set { m_strDisasterCategoryName = value; }
        }

        public string SopName
        {
            get { return m_strSopName; }
            set { m_strSopName = value; }
        }

        public string ActionStepName
        {
            get { return m_strActionStepName; }
            set { m_strActionStepName = value; }
        }

        public string SensorName
        {
            get { return m_strSensorName; }
            set { m_strSensorName = value; }
        }
        
        public string Position
        {
            get { return m_strPosition; }
            set { m_strPosition = value; }
        }

        public DateTime BeginTime
        {
            get { return m_beginTime; }
            set { m_beginTime = value; }
        }

        public DateTime? EndTime
        {
            get { return m_endTime; }
            set { m_endTime = value; }
        }

        public string UserName
        {
            get { return m_strUserName; }
            set { m_strUserName = value; }
        }

        public List<int> AllSensorZoneNos
        {
            get { return m_allSensorZoneNos; }
            set { m_allSensorZoneNos = value; }
        }
    }
}
